### PR TITLE
Update check_dimensions to not raise error for derived dimensions

### PR DESCRIPTION
**Pull request by schymans**
#### References to other Issues or PRs
This addresses the second part of  #16754. 

#### Brief description of what is fixed or changed
check_dimensions was modified to recognise compatible base and derived dimensions, e.g. newton*meter is compatible with joule.

#### Other comments
The purpose of check_dimensions is still not well documented, but it could be very useful.

#### Release Notes

* physics.units
  * Fixed a bug with derived units and dimensions in check_dimensions.

### CODE REVIEW

**Comment by schymans** [comment]
I don't understand why the build still fails. I actually updated the PR and removed the trailing whitespace that travis complained about before, but the commit does not show up here. Can anybode help?

**Comment by oscarbenjamin** [comment]
The complaint is from flake8 and is:
```
+flake8 sympy
sympy/physics/units/util.py:161:5: F401 'functools' imported but unused
sympy/physics/units/util.py:162:5: F401 'operator' imported but unused
```
The complaint is that these functions should not be imported if they are no longer used.

If you install flake8 you can run this command locally as
```
$ flake8 sympy
```

**Comment by schymans** [comment]
Thank you!!

**Comment by schymans** [comment]
@oscarbenjamin @asmeurer, is there anything I should do to get this reviewed and merged?

File: sympy/physics/units/util.py [hunk 1]
```diff
@@ -161,6 +161,15 @@ def check_dimensions(expr, unit_system="SI"):
...
     unit_system = UnitSystem.get_unit_system(unit_system)
 
+    def addDict(dict1, dict2):
+        """Merge dictionaries by adding values of common keys and
+        removing keys with value of 0."""
+        dict3 = {**dict1, **dict2}
+        for key, value in dict3.items():
+            if key in dict1 and key in dict2:
+                   dict3[key] = value + dict1[key]
+        return {key:val for key, val in dict3.items() if val != 0}
```
Comment (Upabjojr) [inline sympy/physics/units/util.py:171]: I was about to suggest using `collections.Counter`, which is a subclass of `dict` for counting, but apparently it doesn't work with negative countings.

**Comment by Upabjojr** [comment]
@schymans ping me for PRs on the units module. I'm currently the only maintainer of this module at SymPy.

**Pull request merged**